### ISSUE TITLE
CSS error given when index.css is used in tutorial

### ISSUE THREAD

**Issue by nsclong** [chromaui/learnstorybook.com#403]
There appears to be a problem with the CSS of the Learn Storybook tutorial.

Following the instructions exactly up to and including the Config section of https://www.learnstorybook.com/intro-to-storybook/react/en/simple-component/ gives the following error:

![image](https://user-images.githubusercontent.com/16524531/97152075-4d8c2580-1768-11eb-8085-61ad38122ea3.png)

I can fix this by commenting out the relevant piece of CSS in `index.css` as follows but then the rendering of the icons is not correct.

![image](https://user-images.githubusercontent.com/16524531/97151954-1fa6e100-1768-11eb-8905-b213d1cf4197.png)

This error may put beginner users off, so any resolution would be much appreciated.

**Comment by jonniebigodes** [comment]
@nsclong sorry for the issue you are facing with the tutorial. I took a look and it seems it might be related to the recent update to `react-scripts` and `react` as part of the `create-react-app` package. As of now when you follow the instructions detailed in the tutorial, your `package.json` will look like this:

```json
{
  "name": "taskbox",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "^5.11.4",
    "@testing-library/react": "^11.1.0",
    "@testing-library/user-event": "^12.1.10",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-scripts": "4.0.0",
    "web-vitals": "^0.2.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject",
    "storybook": "start-storybook -p 6006 -s public",
    "build-storybook": "build-storybook -s public"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@babel/core": "^7.12.3",
    "@storybook/addon-actions": "^6.0.27",
    "@storybook/addon-essentials": "^6.0.27",
    "@storybook/addon-links": "^6.0.27",
    "@storybook/node-logger": "^6.0.27",
    "@storybook/preset-create-react-app": "^3.1.4",
    "@storybook/react": "^6.0.27",
    "babel-loader": "^8.1.0",
    "react-is": "^17.0.1"
  }
}
```
If you continue to follow the instructions provided up to the Simple component page you'll run into the error you mentioned.

As a temporary workaround you could do the following:
- Delete `node_modules` and `yarn.lock`/ `package.lock`.
- Change `react-scripts` to a previous version, you can use `3.43`.
- Change `react` and `react-dom` to a previous version. You can use version `16.3.1`

Which will result in your package.json to look like:
```json
{
  "name": "taskbox",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "^5.11.4",
    "@testing-library/react": "^11.1.0",
    "@testing-library/user-event": "^12.1.10",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "3.4.3",
    "web-vitals": "^0.2.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject",
    "storybook": "start-storybook -p 6006 -s public",
    "build-storybook": "build-storybook -s public"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@babel/core": "^7.12.3",
    "@storybook/addon-actions": "^6.0.27",
    "@storybook/addon-essentials": "^6.0.27",
    "@storybook/addon-links": "^6.0.27",
    "@storybook/node-logger": "^6.0.27",
    "@storybook/preset-create-react-app": "^3.1.4",
    "@storybook/react": "^6.0.27",
    "babel-loader": "^8.1.0",
    "react-is": "^17.0.1"
  }
}
```

And once you issue `yarn storybook` the development build will go through as you can see below:

![working-intro-storybook-cli](https://user-images.githubusercontent.com/22988955/97349809-d77ee000-1887-11eb-8d34-4456c0f3dc3e.png)

And your Storybook will load without issues:

![working-intro-storybook-ui](https://user-images.githubusercontent.com/22988955/97349867-e82f5600-1887-11eb-8161-cf96d74fb9ae.png)

I've checked with the Storybook maintainers and they are actively working on fixing this issue and provide the exact seamless integration that you and the other readers of the tutorial had in the past. 

If you want to keep track on the progress being done. Check the [Storybook Discord server](https://discord.gg/UUt2PJb)  and [repo](https://github.com/storybookjs/storybook). 

Stay safe

**Comment by apontejosea** [comment]
Thanks for the temporary workaround @jonniebigodes.  Back in business.

**Comment by jonniebigodes** [comment]
@apontejosea no need to thank. Glad that i was able to help you. Hopefully this will be fixed soon with 6.1. The maintainers are hard at work trying to release it soon. When it does you won't this workaround anymore. React 17 and CRA recent releases threw a "curve ball" (pardon the bad pun) and it broke some stuff in Storybook.

**Comment by philipmjohnson** [comment]
This fixed it for me. However, there's a small typo in the instructions:  

```
Change react-scripts to a previous version, you can use 3.43.
```

should be:

```
Change react-scripts to a previous version, you can use 3.4.3.
```

**Comment by jonniebigodes** [comment]
@philipmjohnson this is just a minimal set of instructions to provide a workaround. This issue is short term lived and will be closed with the release of Storybook version 6.1. Glad that you managed to get it working. 

Stay safe

**Comment by eddsaura** [comment]
@jonniebigodes thanks for the workaround!

**Comment by nicc777** [comment]
I tried the `Learn Storybook tutorial` today and had to make another change to get going (still on the initial page). 

The command `yarn test --watchAll` was failing with the following error:

```bash
 FAIL  src/App.test.js
  ✕ renders learn react link (1ms)

  ● renders learn react link

    ReferenceError: React is not defined

      4 | 
      5 | test('renders learn react link', () => {
    > 6 |   render(<App />);
        |          ^
      7 |   const linkElement = screen.getByText(/learn react/i);
      8 |   expect(linkElement).toBeInTheDocument();
      9 | });

      at Object.<anonymous> (src/App.test.js:6:10)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        0.87s, estimated 1s
Ran all test suites.
```

I had to modify the following files to get it to work:

In `App.test.js` and `App.js` add the following line right at the top:

```javascript
import React from 'react';
```
After that, everything was starting up what appears to be "normal" - will be trying the rest of the tutorial now.

**Comment by jonniebigodes** [comment]
@nicc777 React 17 and CRA were updated recently and that introduced some changes that were tracked and should now be fixed. Also Storybook just released 6.1 and it should take care of some of the issues that the tutorial has been suffering. I'm currently working on checking the tutorial and see what needs to be changed. 

Stay safe.

**Comment by jonniebigodes** [comment]
Going to close this issue as with the recent merge of #427 the way styling is handled is now fixed. 

Stay safe all
